src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,295 @@
+            } else if (input.startsWith("unmark")) {
+                unmarkInputChecker(input);
+                int taskNum = Integer.parseInt(input.split(" ")[1]);
+                taskList.markTaskAsIncomplete(taskNum);
+                ui.markTaskAsIncompleteMessage(taskList.getTask(taskNum));
+            } else if (input.startsWith("delete")) {
+                deleteInputChecker(input);
+                int taskNum = Integer.parseInt(input.split(" ")[1]);
+                Task toDelete = taskList.getTask(taskNum);
+                taskList.deleteTaskFromList(taskNum);
+                ui.deletedTaskMessage(toDelete, taskList.numberOfTasks());
+            } else if (isEventTask(input)) {
+                eventInputChecker(input);
+                String[] eventConstructor = input.replace("event ", "").split("/at ");
+                String timeModified = eventConstructor[1].replace("from ", "");
+                inputEvent(eventConstructor[0], timeModified);
+            } else if (isDeadlineTask(input)) {
+                deadlineInputChecker(input);
+                addDeadlineFormatted(input);
+            } else if (isTodoTask(input)) {
+                todoInputChecker(input);
+                inputTodo(input.replace("todo ", ""));
+            } else if (isFindTask(input)) {
+                findTaskInputChecker(input);
+                findTasks(input);
+            } else {
+                throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (DukeException e) {
+            ui.printMessage(e.getMessage());
+        }
+    }
+
+    /**
+     * Checks if the input is trying to call a search function.
+     * @param s the input string.
+     * @return true if input describes a query function, else false.
+     */
+    public boolean isFindTask(String s) {
+        return s.startsWith("find");
+    }
+
+    /**
+     * Checks if the input is describing a Deadline object.
+     * @param s The input string.
+     * @return true if input describes a Deadline object, else false.
+     */
+    public boolean isDeadlineTask(String s) {
+        return s.startsWith("deadline");
+    }
+
+    /**
+     * Checks if the input is describing a Todo object.
+     * @param s The input string.
+     * @return true if input describes a Todo object, else false.
+     */
+    public boolean isTodoTask(String s) {
+        return s.startsWith("todo");
+    }
+
+    /**
+     * Checks if the input is describing an Event object.
+     * @param s The input string.
+     * @return true if input describes an Event object, else false.
+     */
+    public boolean isEventTask(String s) {
+        return s.startsWith("event");
+    }
+
+    /**
+     * Checks if the input for a Todo task is valid.
+     * @param input The input string.
+     * @throws DukeException if the input is invalid.
+     */
+    public void todoInputChecker(String input) throws DukeException {
```

Comment:
I noticed some of your methods implementation in this file do not quite align with its semantic meaning, one might expect a boolean value to be returned by reading the javadocs. 
The name for the method as a noun may have circumvented this but it feels weird (using exceptions as a control flow) to have a method whose sole purpose is to throw errors. Perhaps an alternative approach would be to simply throw an error in the main execution if the user input evaluates to false for all your 'checkers'?